src/NUnitFramework/nunitlite/OutputManager.cs

```diff
@@ -120,7 +120,7 @@ public void WriteTestFile(ITest test, OutputSpecification spec)
                         "spec");
             }
 
-            outputWriter.WriteTestFile(test, outputPath);
+            outputWriter.WriteTestFile(test, filter, outputPath);
             Console.WriteLine("Tests ({0}) saved as {1}", spec.Format, outputPath);
         }
     }
```

Comment:
An example of what I'm confused by. Can't we just pass a pre-filtered test hierarchy to the write methods? If it becomes the job of the output writers to do the filtering, then every output writer will have to support it, including those that are implemented as extensions to the engine. And in the case of the engine, we will be changing a public API.